### DISCUSSION TITLE
Bad compilation of ww3_shel in ww3 v7.14

### DISCUSSION THREAD

**Discussion by JMSnyder1980** [NOAA-EMC/WW3#896] [General]
took out the netcdf dependent SCRIPNC from the switch_NCEP_gwm file in WW3/model/bin directory

I ran:

cmake .. -DNETCDF=OFF -DEXCLUDE_FIND="netcdf" -DSWITCH=/home/ww3/WW3/model/bin/switch_NCEP_gwm -DCMAKE_INSTALL_PREFIX=install

and make install

I found in WW3/build/bin executables and when I ran ww3_shel from there I got the following error:

[ww3@wavewatch bin]$ ./ww3_shel

                 *** WAVEWATCH III Program shell ***      
           ===============================================
At line 1195 of file /home/ww3/WW3/model/src/ww3_shel.F90 (unit = 11, file = 'fort.11')
Fortran runtime error: End of file

Error termination. Backtrace:
#0 0x7fdd41449171 in ???
https://github.com/NOAA-EMC/WW3/issues/1 0x7fdd41449d19 in ???
https://github.com/NOAA-EMC/WW3/issues/2 0x7fdd4144a521 in ???
https://github.com/NOAA-EMC/WW3/pull/3 0x7fdd416521bb in ???
https://github.com/NOAA-EMC/WW3/pull/4 0x7fdd416527b2 in ???
https://github.com/NOAA-EMC/WW3/pull/5 0x7fdd4164f33b in ???
https://github.com/NOAA-EMC/WW3/issues/6 0x7fdd41653ec4 in ???
https://github.com/NOAA-EMC/WW3/issues/7 0x7fdd41654d34 in ???
https://github.com/NOAA-EMC/WW3/issues/8 0x402570 in w3shel
at /home/ww3/WW3/model/src/ww3_shel.F90:1195
https://github.com/NOAA-EMC/WW3/pull/9 0x401e4c in main
at /home/ww3/WW3/model/src/ww3_shel.F90:253

This type of output indicates that the compilation did not go through properly for ww3_shel.

I did see this warning in the compilation:

[ 55%] Building Fortran object model/src/CMakeFiles/ww3_lib.dir/w3srcemd.F90.o
/home/ww3/WW3/model/src/w3srcemd.F90:1675:0:

#endif W3_PDLIB

Warning: extra tokens at end of #endif directive

Can you help me resolve this issue?

**Comment by ukmo-ccbunney** [comment]
I don't think the error is related to that `#endif` directive. I believe that is a known (benign) issue.

There is not much to diagnose the error in your output, but I will have a guess...

Unit number 11 is connected to one of the field input files for your model. In this case I think it is the water levels input file.
Have you requested water levels in your ww3_shel.inp file? If so, have you provided a lev.ww3 file with data in it? The error suggests that the WW3 shel is running out of data in reading that file...

Perhaps you can provide your ww3_shel.inp/nml file and a list of your input files?

**Comment by JMSnyder1980** [comment]
Attached are my input files:


[ww3_shel.inp.txt](https://github.com/NOAA-EMC/WW3/files/10758946/ww3_shel.inp.txt)
[ww3_prep.inp.txt](https://github.com/NOAA-EMC/WW3/files/10758949/ww3_prep.inp.txt)
[ww3_grid.inp.txt](https://github.com/NOAA-EMC/WW3/files/10758951/ww3_grid.inp.txt)


How do I generate a lev.ww3 file?

**Reply by aronroland** [reply]
@JMSnyder1980 u have 2 use ww3_prnc

**Reply by JMSnyder1980** [reply]
How do I compile ww3_prnc?   I have tried to compile ww3 based on what is said in manual and quick start guide.   Have not gotten ww3_prnc.   My main goal is to get a success compilation of ww3_shel.  This did NOT happen with the input files I got.   I get the following output for ww3_shel.inp:

                     *** WAVEWATCH III Program shell ***      
               ===============================================

  Comment character is '$'

At line 1205 of file /home/ww3/WW3/model/src/ww3_shel.F90 (unit = 11, file = './ww3_shel.inp')
Fortran runtime error: Bad repeat count in item 2 of list input

Error termination. Backtrace:
#0  0x7f8bc3f8f171 in ???
#1  0x7f8bc3f8fd19 in ???
#2  0x7f8bc3f90521 in ???
#3  0x7f8bc418e3fa in ???
#4  0x7f8bc418e468 in ???
#5  0x7f8bc4190f5a in ???
#6  0x7f8bc41926d9 in ???
#7  0x40264f in w3shel
	at /home/ww3/WW3/model/src/ww3_shel.F90:1205
#8  0x401e4c in main
	at /home/ww3/WW3/model/src/ww3_shel.F90:253
[ww3@wavewatch work_7.14]$ ~/WW3/build/bin/ww3_shel 

                     *** WAVEWATCH III Program shell ***      
               ===============================================

  Comment character is '$'

At line 1205 of file /home/ww3/WW3/model/src/ww3_shel.F90 (unit = 11, file = './ww3_shel.inp')
Fortran runtime error: Bad repeat count in item 2 of list input

Error termination. Backtrace:
#0  0x7fbdeeff5171 in ???
#1  0x7fbdeeff5d19 in ???
#2  0x7fbdeeff6521 in ???
#3  0x7fbdef1f43fa in ???
#4  0x7fbdef1f4468 in ???
#5  0x7fbdef1f6f5a in ???
#6  0x7fbdef1f86d9 in ???
#7  0x40264f in w3shel
	at /home/ww3/WW3/model/src/ww3_shel.F90:1205
#8  0x401e4c in main
	at /home/ww3/WW3/model/src/ww3_shel.F90:253


Can I resolve this running ww3_prnc?   If so how do I compile this?   I need a specific step by step guide in doing so?   However based on the above output and previous experience working with this model and similar ones there is something going on with the code?  Do I have to tweak anything in the code?  If so where?

**Reply by aliabdolali** [reply]
if you have nc-config on your computer, you can:
`export NETCDF_CONFIG=<path to nc-config>`
and then you can use `w3_make` to compile the ww3_prnc program
The aforementioned method is our old gnu build.
Instrad, you can use cmake build following the [instruction](https://github.com/NOAA-EMC/WW3/tree/develop/model#readme)

**Reply by JMSnyder1980** [reply]
Thank you for sending the instructions.  I was able to get cmake to work with netcdf path but it says I am missing netcdf.mod.  How do I obtain this module?
